src/components/comment/comment.scss

```diff
@@ -23,6 +23,7 @@
 
 /// Display a vertical line if the comment contains a reply thread to connect
 /// those comments visually with their parent.
+.c-comment.is-replying::after,
 .c-comment--thread::after {
```

Comment:
I added a separate class that does the same thing as `--thread` so our JS doesn't have to handle correctly toggling `--thread` and account for whether a comment already has a thread or not.